launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java

```diff
@@ -232,20 +232,20 @@ public void run() {
         };
         ServerConnection clientConnection = new ServerConnection(client, timeout);
         Thread clientThread = factory.newThread(clientConnection);
-        synchronized (timeout) {
```

Comment:
we're no longer synchronizing on the `timeout` here, but I didn't see anywhere else synchronizing on it either (including `ServerConnection`).  Given it doesn't escape this method, I'm not sure how multiple threads could ever access `timeout` at once, so it makes sense to remove this synchronization

Reply:
This was exactly my reasoning. I couldn't find any reason why the synchronisation was needed. The removal was only a means to make the code simpler removing the cognitive cost suggested by the presence of a synchronise keyword. The actual fix doesn't depend on it.